demisto_sdk/commands/common/git_util.py

```diff
@@ -1,30 +1,34 @@
 import os
 import re
 from pathlib import Path
-from typing import Set, Tuple, Union
+from typing import Optional, Set, Tuple, Union
 
 import click
 import gitdb
-from git import InvalidGitRepositoryError, Repo
+from git import InvalidGitRepositoryError, Repo  # noqa: TID251
 from git.diff import Lit_change_type
 from git.remote import Remote
 
 from demisto_sdk.commands.common.constants import PACKS_FOLDER
 
 
 class GitUtil:
-    repo: Repo
+    # in order to use Repo class/static methods
+    REPO_CLS = Repo
 
-    def __init__(self, repo: Repo = None):
-        if not repo:
-            try:
-                self.repo = Repo(Path.cwd(), search_parent_directories=True)
-            except InvalidGitRepositoryError:
-                raise InvalidGitRepositoryError(
-                    "Unable to find Repository from current working directory - aborting"
-                )
-        else:
-            self.repo = repo
+    def __init__(
+        self,
+        path: Optional[Path] = None,
+        search_parent_directories: bool = True,
+    ):
+        try:
+            self.repo = Repo(
+                path or Path.cwd(), search_parent_directories=search_parent_directories
+            )
+        except InvalidGitRepositoryError:
+            raise InvalidGitRepositoryError(
+                "Unable to find Repository from current working directory - aborting"
+            )
```

Comment:
```suggestion
        except InvalidGitRepositoryError:
            logger.exception("Unable to find Repository from current working directory - aborting")
        raise
```

Reply:
also, why `from current working directory`? we can do `path = path or Path.cwd()` and then use `path.resolve()` in the error message`

Reply:
agreed